.github/workflows/deploy-docs.yml

```diff
@@ -1,6 +1,13 @@
 name: Publish Docs
 
-on: ["push", "pull_request"]
+on:
+  pull_request:
+  push:
+    branches:
+      - main
```

Comment:
These will not do much apart from checking if sphinx did not fail as the last step will run only on release. But the lines below are needed here.